sdk/src/main/java/com/mapbox/maps/NativeObserver.kt

```diff
@@ -43,13 +41,19 @@ internal class NativeObserver(
       MapEvents.MAP_IDLE -> onMapIdleListeners.forEach { it.onMapIdle() }
       MapEvents.MAP_LOADING_ERROR -> if (onMapLoadErrorListeners.isNotEmpty()) {
         val loadingError = event.getMapLoadingErrorEventData()
+        onStyleLoadingFinishedListeners.clear()
```

Comment:
That should work now because `onStyleLoadingFinishedListeners` and `onMapLoadErrorListeners` are managed internally by us. But if we expose it to user some day - it will not work correctly.
Can we just use single `onStyleLoadingFinishedListener` and `onMapLoadErrorListener`? In current conditions it seems array is useless.

Reply:
> But if we expose it to user some day

We do expose it to users and plugins already:
```
MapboxMap#addOnStyleLoadingFinishedListener
```

> it will not work correctly.

Is it because you expect a listener to get continous style load updates versus 1 shot invocation? This is indeed maybe not correct and needs to be altered to only allow Style.OnStyleLoaded to be invoked once but all `onStyleLoadingFinishedListeners` should be invoked always. Going to revisit this. 

> In current conditions it seems array is useless.

We can have multiple components listening to this (ranging from users to plugin to our internal style loaded callback system).